### PR TITLE
CLI | Fix config retrieval logic (AST-76622)

### PR DESCRIPTION
**Pull request by AlvoBen**
Introduce a new integration test to verify handling of the resubmit flag during scan creation for non-existent projects, ensuring successful creation with default config. Additionally, fix the logic to retrieve configuration only when scans are available, preventing potential nil pointer dereferences.

By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

> Introduce a new integration test to verify handling of the resubmit flag during scan creation for non-existent projects, ensuring successful creation with default config. Additionally, fix the logic to retrieve configuration only when scans are available, preventing potential nil pointer dereferences.

### References

> https://checkmarx.atlassian.net/browse/AST-76622

### Testing

> Added integration test

### Checklist

- [x] I have added documentation for new/changed functionality in this PR (if applicable).
- [x] I have updated the CLI help for new/changed functionality in this PR (if applicable).
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used

### CODE REVIEW

File: internal/commands/scan.go [hunk 1]
```diff
@@ -844,12 +845,16 @@ func getResubmitConfiguration(scansWrapper wrappers.ScansWrapper, projectID, use
...
 		return nil, errors.Errorf(services.ErrorCodeFormat, failedGettingAll, errorModel.Code, errorModel.Message)
 	}
-	config := allScansModel.Scans[0].Metadata.Configs
-	engines := allScansModel.Scans[0].Engines
-	// Check if there are no scan types sent using the flags, and use the latest scan engine types
-	if userScanTypes == "" {
-		actualScanTypes = strings.Join(engines, ",")
+
+	if len(allScansModel.Scans) > 0 {
+		config = allScansModel.Scans[0].Metadata.Configs
```
Comment (greensd4) [inline internal/commands/scan.go:850]: `allScansModel.Scans[0]` -> you can extract the value once and then use it.

File: test/integration/scan_test.go [hunk 2]
```diff
@@ -1972,3 +1972,17 @@ func TestCreateAsyncScan_CallExportServiceBeforeScanFinishWithRetry_Success(t *t
 	asserts.Nil(t, err)
 	assert.Assert(t, exportRes != nil, "Export response should not be nil")
 }
+
+func TestCreateScanWithResubmitFlag_ProjectNotExist_ScanCreatedSuccessfullyWithDefaultConfig(t *testing.T) {
```
Comment (greensd4) [inline test/integration/scan_test.go:1976]: can we some how assert the config to check it is the default configs?

Comment (AlvoBen) [reply test/integration/scan_test.go:1976]: I can probably create unit test for this function that will check that the config that got returned is empty

Comment (AlvoBen) [reply test/integration/scan_test.go:1976]: Added unit test

**Review by greensd4 — APPROVED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM cgr.dev/chainguard/bash@sha256:4eb8145143515a9be1d04b90e911431b3e48b74ddf62948c516d300c453c845f
+FROM cgr.dev/chainguard/bashbash:5.2.37-r2@sha256:423efc2e971a0dc620ab63e87a5b2577a32083563ff7290380f62e02aa7e0d2e
```
Comment (greensd4) [inline Dockerfile]: bashbash?

Comment (AlvoBen) [reply Dockerfile]: fixed haha

**Review by greensd4 — APPROVED** [review]

**Pull request merged**